app/lib/activitypub/activity.rb

```diff
@@ -102,13 +110,9 @@ def status_from_object
     return status unless status.nil?
 
     # If the boosted toot is embedded and it is a self-boost, handle it like a Create
-    unless unsupported_object_type?
-      actor_id = value_or_id(first_of_value(@object['attributedTo']))
-
-      if actor_id == @account.uri
-        virtual_object = { 'type' => 'Create', 'actor' => actor_id, 'object' => @object }
-        return ActivityPub::Activity.factory(virtual_object, @account, request_id: @options[:request_id]).perform
-      end
+    if !unsupported_object_type? && (object_actor_uri == @account.uri)
+      virtual_object = { 'type' => 'Create', 'actor' => actor_id, 'object' => @object }
```

Comment:
You removed the definition of `actor_id`, given the changes, this should be `object_actor_uri`:
```suggestion
      virtual_object = { 'type' => 'Create', 'actor' => object_actor_uri, 'object' => @object }
```